gpii/node_modules/flowManager/src/FlowManagerUtilities.js

```diff
@@ -31,14 +31,17 @@
         };
     };
 
-    gpii.request.flowManager.getPreferences = function (preferencesDataSource, event, userToken) {
+    gpii.request.flowManager.getPreferences = function (preferencesDataSource, request, event, userToken) {
         fluid.log("gpii.request.flowManager.getPreferences called - fetching preferences");
-        preferencesDataSource.get({
-            userToken: userToken
-        }, gpii.request.flowManager.logAndNotify("Fetched user preferences: ", event,
-            function getPreferences(data) {
-                return data.preferences || data;
-            }));
+        var promise = preferencesDataSource.get({userToken: userToken});
+        promise.then(function (data) {
+            if (data.isError) {
+                request.events.onError.fire(data);
+            }
+            event.fire(data.preferences || data);
+        }, function (err) {
+            request.events.onError.fire("Rejected preference server promise to preferences server. ", err.message);
```

Comment:
This error message is unintelligible - see below - promises don't actually exist anyway